### ISSUE TITLE
Treats markdown link text as URL

### ISSUE THREAD

**Issue by dbr** [lycheeverse/lychee#1579]
If I have a markdown where the link text is a broken link, but the actual link target is a valid URL,

```
$ cat test.md 
[https://dummyexample.gov/notexist (archive.org link)](https://web.archive.org/web/20241129184733/http://example.com)
```

I would expect it only checks the web.archive.org link, but it checks both the link-text and the URL

```
$ lychee -v test.md 
   [ERROR] https://dummyexample.gov/notexist
     [200] https://web.archive.org/web/20241129184733/http://example.com

Issues found in 1 input. Find details below.

[test.md]:
   [ERROR] https://dummyexample.gov/notexist

🔍 2 Total (in 2s) ✅ 1 OK 🚫 1 Error
```

This seems to relate to https://github.com/lycheeverse/lychee/issues/949#issuecomment-2400908728 except this is a slightly different case (as a markdown parser might helpfully convert `http://example.com` to `<a href="example.com">http://example.com</a>` - but it would surely be nonsensical to ever do the same for `dummyexample.gov` in the initial example - it would be a link inside a link)

**Comment by mre** [comment]
Yeah, we started treating plaintext links in HTML files as false-positives in https://github.com/lycheeverse/lychee/pull/1528. Something similar could work for Markdown files.

@markogalevski fyi

**Comment by thomas-zahner** [comment]
@mre I guess this is related to #1399? Should the text of  a Markdown link only be checked with the `--include-verbatim` option as mentioned there?

**Comment by mre** [comment]
Yes, that sounds like the way to go! Thanks for recognizing the connection.

**Comment by thomas-zahner** [comment]
@mre So to recap; "verbatim" for Markdown and HTML seem to have different meanings.

With HTML we only detect links as such if we can be sure that they are supposed to be valid links, meaning we ignore links form "plain text" that could be links.

```html
<p>This is considered verbatim and not found unless the flag is specified: https://example.com</p>
```

With Markdown we are less strict:

```markdown
This is not considered verbatim, meaning it is found even without the flag: https://example.com

only things like this are considered verbatim: `https://example.com`
```

I think we should give verbatim the same meaning for Markdown files as with HTML files. Do you agree? If so, I will try to change this behaviour and update the documentation accordingly, as the current documentation is not very clear about the meaning of verbatim. This will of course be a breaking change. It will prevent false positives like in this issue at the cost of detecting fewer links as long as `--include-verbatim` is not specified.

**Comment by mre** [comment]
Yes.

If it helps, you might want to use this issue to sketch out the planned changes so that we can sign off on them. That might save you some time down the road and could serve as documentation. I think we should definitely document the behavior better as well. 👍

**Comment by mre** [comment]
Should be fixed in `master` now and will be released with the next version.
